### PR TITLE
ci: Add GitHub action that generates typing and commits it

### PR DESCRIPTION
**Pull request by chamini2**
Problem: https://github.com/stefanzweifel/git-auto-commit-action#commits-made-by-this-action-do-not-trigger-new-workflow-runs

> Commits made by this Action do not trigger new Workflow runs
> 
> The resulting commit will not trigger another GitHub Actions Workflow run. This is due to [limitations set by GitHub](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow).
> 
>     When you use the repository's GITHUB_TOKEN to perform tasks on behalf of the GitHub Actions app, events triggered by the GITHUB_TOKEN will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.
> 
> You can change this by creating a new [Personal Access Token (PAT)](https://github.com/settings/tokens/new), storing the token as a secret in your repository and then passing the new token to the [actions/checkout](https://github.com/actions/checkout#usage) Action step.
> 
> ```
> - uses: actions/checkout@v2
>   with:
>     token: ${{ secrets.PAT }}
> ```
> 
> If you create a personal access token, apply the repo and workflow scopes.
> 
> If you work in an organization and don't want to create a PAT from your personal account, we recommend using a [robot account](https://docs.github.com/en/github/getting-started-with-github/types-of-github-accounts) for the token.

<details>
closes FEA-358
</details>

### CODE REVIEW

File: tools/generate_typing_context.py [hunk 1]
```diff
@@ -84,7 +85,9 @@ def generate_protocols(file, class_name):
         if ast.get_docstring(call_function):
             call_function.body = call_function.body[:1]
         else:
-            call_function.body = [ast.Expr(ast.Constant(...))]
+            call_function.body = []
+
+        call_function.body.append(ast.Expr(ast.Constant(...)))
```
Comment (isidentical) [inline tools/generate_typing_context.py:90]: ATL: why do we need this?

**Review by isidentical — COMMENTED** [review]
Can we make it a PR, to have some human interaction.

File: tools/generate_typing_context.py (hunk 1 above)
Comment (chamini2) [reply tools/generate_typing_context.py:90]: image

**Comment by chamini2** [comment]
> Can we make it a PR, to have some human interaction.

I don't think it's necessary this commits to the PR you are doing, not to the `main` branch.

image

Look how I pushed and then the github-actions user pushed the new typing

File: tools/generate_typing_context.py (hunk 1 above)
Comment (isidentical) [reply tools/generate_typing_context.py:90]: Seems like a bug in pylance TBH (protocols are not supposed return a thing). But if this makes it work, perfect 💯

**Comment by isidentical** [comment]
Ah, I didn't notice it pushed to your PR. Then LGTM 🚀

**Review by isidentical — APPROVED** [review]

**Pull request merged**